services/streammanager/googlepubsub/googlepubsubmanager.go

```diff
@@ -60,7 +60,7 @@ func NewProducer(destinationConfig interface{}, o Opts) (*PubsubClient, error) {
 		return nil, fmt.Errorf("invalid configuration provided, missing projectId")
 	}
 	var client *pubsub.Client
-	if config.Credentials != "" { // Normal configuration requires credentials
+	if !googleutils.IsEmptyCredentials(config.Credentials) { // Normal configuration requires credentials
```

Comment:
The client initialization flow is not consistent across the destinations. It'd be easier with the introduction of the server env variable now. I'd imagine the flow to be something like the below:

```
// Server environment variable takes precedence 
// in controlling the flow
if (<server env> != true) {
  // validate the creds and throw error on failure
}
// initialize the client
```

Reply:
Refactored this, kindly check and confirm if it is ok

Reply:
I didn't understand the new changes completely. Are we not going to use the environment variables?
Moreover, the flow still seems to be inconsistent across different destinations.

Reply:
The environment variables check is abstracted into a method which is defined in `googleutils.go`. You can check [here](https://github.com/rudderlabs/rudder-server/blob/53a4c1a53030c9114559552c25231fca50751994/utils/googleutils/googleutils.go#L22).
What flow are you talking about here ?

Reply:
The client initialisation flow is fairly common in all the google destinations(now atleast). Can you check [this](https://github.com/rudderlabs/rudder-server/blob/53a4c1a53030c9114559552c25231fca50751994/services/streammanager/bqstream/bqstreammanager.go#L74-L81) and let me know if I'm missing something here ?

Reply:
I meant, the variables are defined in the configuration file, config.yaml. Isn't the idea to drive it from the deployment YAML file?

Reply:
Got it, but we should remove them before merging the changes.